### PR TITLE
Add auth provider

### PR DESCRIPTION
**Pull request by yash9263**

### CODE REVIEW

File: src/hooks/useForm.ts [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+const useForm = (initialState) => {
+  const [values, setValues] = useState(initialState)
+  //   TODO add an error state to validate all fields and store the error for them
+  const handleChange = (e) => {
+    let { name, value, type } = e.target
+    if (type === 'number') {
+      value = parseInt(value)
+    }
+
+    setValues((prevValues) => ({ ...prevValues, [name]: value }))
```
Comment (Dev-AkshatGupta) [inline src/hooks/useForm.ts:12]: I want to know what you have done in this while putting the square braces around name what's this for?

File: src/hooks/useLocalStorage.ts [hunk 2]
```diff
@@ -0,0 +1,21 @@
+import { useState } from 'react'
+
+// reference: https://usehooks.com/useLocalStorage
+const useLocalStorage = <T>(key: string) => {
```
Comment (Dev-AkshatGupta) [inline src/hooks/useLocalStorage.ts:4]: So are you using typeScript??

File: src/routes/signup.tsx [hunk 3]
```diff
@@ -0,0 +1,76 @@
...
+            required
+          />
+        </div>
+        <div>
+          <label>password</label>
+          <input
+            name="password"
+            type="password"
+            id="signup-password"
+            value={formValues.password}
```
Comment (Dev-AkshatGupta) [inline src/routes/signup.tsx:63]: this is good that you are maintaining the value attributes for each input this is really good practice I will adopt this

File: src/server.js [hunk 4]
```diff
@@ -35,7 +35,7 @@ export function makeServer({ environment = 'development' } = {}) {
     seeds(server) {
       // disballing console logs from Mirage
 
-      server.logging = false
+      server.logging = true
```
Comment (Dev-AkshatGupta) [inline src/server.js:38]: So you basically changed the backend a bit for the login or what ?

**Review by Dev-AkshatGupta — APPROVED** [review]
Cool code Got to learn so much from your coding style

**Review by vrx29 — APPROVED** [review]

File: src/hooks/useLocalStorage.ts (hunk 2 above)
Comment (yash9263) [reply src/hooks/useLocalStorage.ts:4]: not very good effectively 😅

File: src/routes/signup.tsx (hunk 3 above)
Comment (yash9263) [reply src/routes/signup.tsx:63]: cool 🙌

File: src/server.js (hunk 4 above)
Comment (yash9263) [reply src/server.js:38]: nope it's just to log the requests that I made to the mock server you can read about it more on miragejs docs [click here](https://miragejs.com/api/classes/server/#logging)

File: src/index.tsx [hunk 5]
```diff
@@ -4,15 +4,18 @@ import { BrowserRouter } from 'react-router-dom'
...
 
 ReactDOM.render(
   <React.StrictMode>
-    <BrowserRouter>
-      <GlobalStyles />
-      <App />
-    </BrowserRouter>
+    <AuthProvider>
+      <BrowserRouter>
+        <GlobalStyles />
```
[the remark this answers is not in the document]
Comment (yash9263) [reply src/index.tsx:15]: yep It's supposed to be like this it's styled-components thing [docs](https://styled-components.com/docs/api#createglobalstyle)

File: src/routes/login.tsx [hunk 6]
```diff
@@ -0,0 +1,59 @@
...
+const LoginRoute = () => {
+  const [formValues, handleChange] = useForm(initialValues)
+  const auth = useAuth()
+
+  const handleSubmit: HandleSubmit = async (e) => {
+    if (!e) return
+    e.preventDefault()
+    // TODO handle form validation
+
+    await auth.login({ ...formValues })
```
[the remark this answers is not in the document]
Comment (yash9263) [reply src/routes/login.tsx:22]: yep will do that

File: src/index.tsx (hunk 5 above)
[the remark this answers is not in the document]
Comment (vrx29) [reply src/index.tsx:15]: ok cool then

**Pull request merged**